NEWS.md

```diff
@@ -520,11 +520,12 @@ Deprecated or removed
     `Meta.lower(module, ex)` ([#22064, #24278]).
 
   * `ones(A::AbstractArray[, opts...])` and `zeros(A::AbstractArray[, opts...])` methods
-    have been deprecated. The general replacement is `fill!(similar(A[, opts...]), {1|0})`,
+    have been deprecated. The general replacement is
+    `fill!(similar(A[, opts...]), {one(eltype(A)) | zero(eltype(A))})`,
     though in most use cases simpler alternatives are better: For `zeros(A)`, consider
     `zero(A)`. For `ones(A)` or `zeros(A)`, consider `fill(v, size(A))` for `v` an
-    appropriate one or zero, `fill!(copy(A), {1|0})`, `ones(size(A))` or
-    `zeros(size(A))`, or any of the preceding with different element type
+    appropriate one or zero, `fill!(copy(A), {one(eltype(A)) | zero(eltype(A))})`,
```

Comment:
Perhaps leave `fill!(copy(A), {1|0})` where it is, add `fill!(similar(A), {1|0})` shortly thereafter, and only later suggest `fill!(copy(A), {one(eltype(A)) | zero(eltype)})` for where strictly necessary? Providing the simplest possible replacements that will work in most cases first seems best.